File: duxy/Day1/ej2.java
import java.io.*;
import java.io.*;
import java.io.File;
import java.util.Scanner;
class ej2 {
    public static void main(String[] args) {  
        String line;  
        String all = "";
        String[] parts = null;
        int addition = 0;
        try {            
            Scanner input = new Scanner(System.in);
            File file = new File("input.txt");
            input = new Scanner(file);
            while (input.hasNextLine()) {
                 all = all + input.nextLine();
            }
            input.close();  
        } catch (IOException e) {
            e.printStackTrace();
        }
        parts = all.split("");
        int displa = parts.length/2;
        for(int i = 0; i < parts.length; i++){
            int index = (((i + displa)% parts.length));
            if (parts[i].equals(parts[index])){
               addition = addition + Integer.parseInt(parts[i]); 
            }               
        }            
       System.out.println(addition);
    }
}

File: duxy/Day5/ej2.js
var fs = require('fs');
var index;

fs.readFile("input.txt", 'utf8', function(err, data) {
  if (err) throw err;  
  code(data);

});
function code(data){  
  index = 0;
  data = data.split("\n");  
  while(index < data.length){
    data[index] = parseInt(data[index]);
    index = index + 1
  }
  var index = data[0];
  var indan = 0;
  var sum = 0
  while(index < data.length){
    sum = sum + 1;
    indan = index;
    index = index + data[index];
    if (index >= data.length){
       break;
    }
    if(data[indan] > 2){
      data[indan] = data[indan] - 1;
    } else {
      data[indan] = data[indan] + 1;
    }
  }
  console.log(sum);
}